mem/buffer_slice.go

```diff
@@ -0,0 +1,155 @@
+package mem
+
+import (
+	"io"
+)
+
+// BufferSlice offers a means to represent data that spans one or more Buffer
+// instances.
+type BufferSlice []*Buffer
```

Comment:
Also, worth mentioning that a BufferSlice is meant to be immutable, hence methods have value receiver type.